src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/AddAzureRmSqlServerTransparentDataEncryptionCertificate.cs

```diff
@@ -0,0 +1,211 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.Server.Model;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Adapter;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet
+{
+    /// <summary>
+    /// Defines the Add-AzureRmSqlServerTransparentDataEncryptionCertificate cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Add, "AzureRmSqlServerTransparentDataEncryptionCertificate", ConfirmImpact = ConfirmImpact.Low, SupportsShouldProcess = true, DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(bool))]
+    public class AddAzureRmSqlServerTransparentDataEncryptionCertificate : AzureSqlCmdletBase<IEnumerable<AzureRmSqlServerTransparentDataEncryptionCertificateModel>, AzureSqlDatabaseTransparentDataEncryptionArmAdapter>
+    {
+        private const string PrivateBlobHelpText =
+            "The Private blob for Transparent Data Encryption Certificate. For detailed instructions on how to generate the blob go to https://aka.ms/tdecertprep";
+
+        /// <summary>
+        /// Parameter sets
+        /// </summary>
+        protected const string DefaultParameterSet = "AddAzureRmSqlServerTransparentDataEncryptionCertificateDefaultParameterSet";
+        protected const string InputObjectParameterSet = "AddAzureRmSqlServerTransparentDataEncryptionCertificateInputObjectParameterSet";
+        protected const string ResourceIdParameterSet = "AddAzureRmSqlServerTransparentDataEncryptionCertificateResourceIdParameterSet";
+
+        /// <summary>
+        /// Gets or sets the Server Object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The sql server input object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlServerModel InputObject { get; set; }
+
```

Comment:
Does it make sense to name the parameter SqlServer instead of InputObject?

Reply:
Addressed